### ISSUE TITLE
[torchlib][feature request] aten::einsum

### ISSUE THREAD

**Issue by BowenBao** [microsoft/onnxscript#1130]
Similar to `batchnorm`, `addmm` and `softmax` etc, it would be better aesthetically and performance wise to export `aten::einsum` with `onnx::Einsum`.

Example:

```
import torch


def f(x, y):
    softmax_scale = 1.0
    q = x
    k = y
    return torch.einsum("bthd,bshd->bhts", q, k * softmax_scale)


x = torch.randn(1, 2, 3, 4)
y = torch.randn(1, 2, 3, 4)

print(f(x, y))


# torch.onnx.dynamo_export(f, x, y).save_diagnostics("repro_einsum.sarif")
torch.onnx.dynamo_export(f, x, y).save("repro_einsum.onnx")
```

<img width="272" alt="image" src="https://github.com/microsoft/onnxscript/assets/9376104/e55e0825-814b-4d09-a790-631f0cbfcb39">

**Comment by BowenBao** [comment]
Ok, this might be harder than it looks. pytorch make_fx is lowering and decomposing `torch.einsum` all the time. We'd need to switch to `pre_dispatch=True`, but there are other aspects that might be incompatible, e.g., functionalization.

**Comment by justinchuby** [comment]
https://github.com/onnx/onnx/issues/5739
